File: cpp/1500A.cpp
// 1500A. Going Home
// Completed in 592 ms 58700 kb
// 03/27/2021
#include <iostream>
#define newl '\n'
using namespace std;

int main() {
    int n;
    cin >> n;
    
    int nums[5000000], x[5000000], y[5000000];
    for (int i = 0; i < n; i++) {
        cin >> nums[i];
    }
    
    for (int i = 1; i <= n; i++) {
        for (int j = i+1; j <= n; j++) {
            int s = nums[i] + nums[j];
            
            if (x[s]) {
                if(i != x[s] && i != y[s] && j != x[s] && j != y[s]) {
                    cout << "YES" << newl;
                    cout << i << " ";
                    cout << j << " ";
                    cout << x[s] << " ";
                    cout << y[s] << newl;
                    return 0;
                }
            }
            x[s] = i;
            y[s] = j;
        }
    }
    cout << "NO" << newl;
    return 0;
}

File: python/1500A.py
# 1500A. Going Home
# Completed in 1903 ms 122100 KB
# 03/27/2021

__name__ = "__main__"

if __name__ == "__main__":

    n = int(input())
    nums = list(map(int, input().split())) 

    # let's put all of our sums in here
    sums = {}

    solved = False 

    for i in range(0, n):
        for j in range(0, i):

            if nums[i] + nums[j] in sums:
                a, b = sums[nums[i] + nums[j]]
            
                if i != a and i != b and j != a and j != b:
                    print("YES") 
                    print(i+1, j+1, a+1, b+1)
                    solved = True
                    break 

            else:
                # add nums[i] + nums[j] to sums
                sums[nums[i] + nums[j]] = [i, j]
        
        if solved:
            break

    if not solved:
        print("NO")
        
